kngsild/src/main/kotlin/io/egm/kngsild/utils/AuthUtils.kt

```diff
@@ -15,51 +13,65 @@ import java.net.http.HttpRequest
 import java.net.http.HttpResponse
 
 class AuthUtils(
-    private val serverUrl: String,
-    private val clientId: String,
-    private val clientSecret: String,
-    private val grantType: String,
-    private val enabled: Boolean = true
+    private val clientCredentials: ClientCredentials? = null,
+    private val providedToken: ProvidedToken? = null,
+    private val authType: AuthType = AuthType.NONE
 ) {
 
+    init {
+        when (authType) {
+            AuthType.PROVIDED_TOKEN ->
+                if (providedToken == null) throw ConfigurationError("ProvidedToken are not set")
+            AuthType.CLIENT_CREDENTIALS ->
+                if (clientCredentials == null) throw ConfigurationError("ClientCredentials are not set")
+                else this
```

Comment:
badly indented?

Reply:
Yes the `else` is not for the `if` but for the `when`